src/Microsoft.Azure.SignalR/ClientInvocation/CallerClientResultsManager.cs

```diff
@@ -20,28 +21,45 @@ internal sealed class CallerClientResultsManager : ICallerClientResultsManager,
         private long _lastInvocationId = 0;
 
         private readonly IHubProtocolResolver _hubProtocolResolver;
+        private IEndpointRouter _endpointRouter { get; }
+        private IServiceEndpointManager _serviceEndpointManager { get; }
+        private readonly AckHandler _ackHandler = new();
 
-        public CallerClientResultsManager(IHubProtocolResolver hubProtocolResolver)
+        public CallerClientResultsManager(IHubProtocolResolver hubProtocolResolver, IServiceEndpointManager serviceEndpointManager, IEndpointRouter endpointRouter)
         {
             _hubProtocolResolver = hubProtocolResolver ?? throw new ArgumentNullException(nameof(hubProtocolResolver));
+            _serviceEndpointManager = serviceEndpointManager ?? throw new ArgumentNullException(nameof(serviceEndpointManager));
+            _endpointRouter = endpointRouter ?? throw new ArgumentNullException(nameof(endpointRouter));
```

Comment:
endpointRouter is nullable when it's single endpoint, as you rely on this to detect multi ack is needed in line 47...

Reply:
I should have removed detection in L47. `endpointRouter` is non-nullable.